File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/module/FragmentModule.java
package com.mindorks.bootcamp.learndagger.di.module;

import android.content.Context;

import androidx.fragment.app.Fragment;

import com.mindorks.bootcamp.learndagger.di.qualifier.ActivityContext;

import dagger.Module;
import dagger.Provides;

@Module
public class FragmentModule {

    private Fragment fragment;

    public FragmentModule(Fragment fragment){
        this.fragment=fragment;
    }

    @ActivityContext
    @Provides
    Context provideContext(){
        return fragment.getContext();
    }
}

File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/module/FragmentModule.kt
package com.mindorks.bootcamp.learndagger.di.module

import android.content.Context
import androidx.fragment.app.Fragment
import com.mindorks.bootcamp.learndagger.di.qualifier.ActivityContext
import dagger.Module
import dagger.Provides

@Module
class FragmentModule constructor(private val fragment: Fragment) {
    @ActivityContext
    @Provides
    fun provideContext(): Context? = fragment.context
}